moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp

```diff
@@ -555,7 +555,7 @@ class MoveGroupInterface::MoveGroupInterfaceImpl
     return true;
   }
 
-  bool getCurrentState(robot_state::RobotStatePtr& current_state, double wait_seconds = 1.0)
+  bool getCurrentState(robot_state::RobotStatePtr& current_state, double wait_seconds = 1.0) const
```

Comment:
For this one, the situation is a bit less clear and I would like second opinions here.
The method logically does not modify anything by itself, but it *might* start a topic subscriber and run a thread to start updating an internal `RobotState` object (maintained in the CSM).

From my perspective, this should not be considered `const`.

Reply:
I think this falls into the same category as the functions calling ROS actions. Technically it is const function but at the same the internal state of the object pointed to by the member variable might be modified so marking this as const can be considered as misleading and wrong. Feel like grey area but will revert back to non-const.

Reply:
There is a fairly common pattern in our codebase to append 'NonConst' to functions that are intended to be modifiable, such that you have:

getCurrentState()
getCurrentStateNonConst()

Maybe that helps here?

Reply:
I agree with @v4hn here. `const` should only be used for member functions that neither alter the internal state nor the observable state of an object (except for `mutable` caches). In this case if you run `getCurrentState()` with `wait_seconds`=0.0 the result depends on previous calls and if the ROS action/service call was successful. The same could be argued for most functions that rely on ROS messages.